test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -154,6 +154,26 @@
                     :actual actual#})
         actual#))))
 
+;; This additional slingshot helper assumes our :kind/:msg maps and is probably
+;; brittle.
+(defmethod assert-expr 'thrown-with-slingshot-msg? [msg form]
+  (let [expected (nth form 1)
+        body     (nthnext form 2)]
+    `(sling/try+
+      ~@body
+      (do-report {:type :fail :message ~msg :expected ~expected :actual nil})
+      (catch map? actual#
+        (let [result# (if (or (= actual# ~expected)
+                              (and (= (:kind actual#) (:kind ~expected))
+                                   (re-matches (:msg ~expected) (:msg actual#))))
+                        :pass
+                        :fail)]
+          (do-report {:type result#
+                      :message ~msg
+                      :expected ~expected
+                      :actual actual#}))
+        actual#))))
+
```

Comment:
ugh so i assumed we added our own because there *wasn't* something in slingshot, hence adding another custom thing. I'll put a note in the code so others don't do the same thing.